### PR TITLE
fix: A user joining from invite link of a team doesn't automatically become member of the org  

### LINKED ISSUE #12813: [CAL-2812] A user joining from invite link of a team doesn't automatically become member of the org.
Steps:
- Create a team(Engineering) in an org(Org1 from org1.com). Also, verify that organization.
- Copy Invite link for Engineering Team
- Open the link in incognito
- Signup using the link with email [member@org1.com](mailto:member@org1.com).
- Notice that the user doesn't show up as an organization member in settings/organizations/members

Reported here - https://calcominc.slack.com/archives/C05V243E153/p1702407169502219

<sub>[CAL-2812](https://linear.app/calcom/issue/CAL-2812/a-user-joining-from-invite-link-of-a-team-doesnt-automatically-become)</sub>

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

- Add membership of the organization of the team in case of team invite link Fixes #12813
- Fix derived username from email in case of team invite  - For a verified domain of org1.com and email of member@org1.com, the username was wrongly created as `member-org1 `when it should be `member`.
- Also refactored invitation tests for org to abstract out a few things plus added the tests for the above scenarios

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Tests (Unit/Integration/E2E or any other test)

## How should this be tested?
- Create a team(Engineering) in an org(Org1 from org1.com). Also, verify that organization.
- Copy Invite link for Engineering Team
- Open the link in incognito
- Signup using the link with email member@org1.com. 
- Verify that the username is `member` and not `member-org1` 
- Also, verify that the user shows up as an organization member settings/organizations/members

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

**Comment by hariombalhara** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #12735** <a href="https://app.graphite.dev/github/pr/calcom/cal.com/12735?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #12774** <a href="https://app.graphite.dev/github/pr/calcom/cal.com/12774?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/calcom/cal.com/12774?utm_source=stack-comment).

File: apps/web/playwright/organization/organization-invitation.e2e.ts
Comment (hariombalhara) [inline apps/web/playwright/organization/organization-invitation.e2e.ts:1]: Hide Whitespace changes to better review this file.

File: apps/web/playwright/organization/organization-invitation.e2e.ts [hunk 1]
```diff
@@ -13,169 +14,435 @@ test.afterEach(async ({ users, emails }) => {
...
-      await expect(
-        page.locator(`[data-testid="email-${invitedUserEmail.replace("@", "")}-pending"]`)
-      ).toHaveCount(0);
+      await page.waitForLoadState("networkidle");
+
+      await test.step("By email", async () => {
+        const invitedUserEmail = `rick-${Date.now()}@domain.com`;
+        // '-domain' because the email doesn't match orgAutoAcceptEmail
+        const usernameDerivedFromEmail = `${invitedUserEmail.split("@")[0]}-domain`;
+        await inviteAnEmail(page, invitedUserEmail);
```
Comment (hariombalhara) [inline apps/web/playwright/organization/organization-invitation.e2e.ts:30]: Abstraction done everywhere

File: packages/trpc/server/routers/viewer/teams/createInvite.handler.ts [hunk 2]
```diff
@@ -36,7 +31,7 @@ export const createInviteHandler = async ({ ctx, input }: CreateInviteOptions) =
     if (!existingToken) throw new TRPCError({ code: "NOT_FOUND" });
     return {
       token: existingToken.token,
-      inviteLink: await getInviteLink(existingToken.token, isOrg, orgMembers?.length),
+      inviteLink: await getInviteLink(existingToken.token, isOrganizationOrATeamInOrganization),
```
Comment (hariombalhara) [inline packages/trpc/server/routers/viewer/teams/createInvite.handler.ts:34]: cc @Udit-takkar for review. Why do we need orgMembers? Shouldn't this work.

**Review by Udit-takkar — APPROVED** [review]
LGTM.

**Comment by Udit-takkar** [comment]
some UI colors inconsistency ( not related to changes made in this PR)

Screenshot 2023-12-18 at 5 09 47 PM

Screenshot 2023-12-18 at 5 10 05 PM

**Comment by Udit-takkar** [comment]
@hariombalhara one more thing if i invite someone by email(not copy by link) say udit@cal.com.

The username should have been udit-cal instead of udit as my org is acme.com. only the signup page displays it wrong and the username stored in DB is udit-cal

Screenshot 2023-12-18 at 5 11 31 PM

Screenshot 2023-12-18 at 5 14 52 PM

**Comment by hariombalhara** [comment]
Thanks @Udit-takkar. Opened a PR to fix it [here](https://github.com/calcom/cal.com/pull/12854)

**Pull request merged**